build.sbt

```diff
@@ -1,130 +1,156 @@
+      "org.typelevel" %% "cats-effect" % CatsEffectVersion,
+      scalacheck
+    ) ++ (if (scalaVersion.value.startsWith("2.")) {
+            Seq(
+              "org.scala-lang"          % "scala-reflect"           % scalaVersion.value,
+              "org.scala-lang.modules" %% "scala-collection-compat" % "2.6.0"
+            )
+          } else Nil)
+  )
+
+lazy val memcached = createSubModule("memcached")
   .settings(
     libraryDependencies ++= Seq(
       "net.spy" % "spymemcached" % "2.12.3"
     )
   )
 
-lazy val redis = createModule("redis")
+lazy val redis = createSubModule("redis")
   .settings(
     libraryDependencies ++= Seq(
       "redis.clients" % "jedis" % "3.7.1"
-    ),
-    coverageMinimum       := 56,
-    coverageFailOnMinimum := true
+    )
   )
 
-lazy val caffeine = createModule("caffeine")
+lazy val caffeine = createSubModule("caffeine")
   .settings(
     libraryDependencies ++= Seq(
       "com.github.ben-manes.caffeine" % "caffeine"            % "3.0.6",
       "org.typelevel"                %% "cats-effect-testkit" % CatsEffectVersion % Test,
       "com.google.code.findbugs"      % "jsr305"              % "3.0.2"           % Provided
-    ),
-    coverageMinimum       := 80,
-    coverageFailOnMinimum := true
+    )
   )
 
-lazy val circe = createModule("circe")
+lazy val circe = createSubModule("circe")
   .settings(
     libraryDependencies ++= Seq(
       "io.circe" %% "circe-core"    % "0.14.1",
       "io.circe" %% "circe-parser"  % "0.14.1",
       "io.circe" %% "circe-generic" % "0.14.1" % Test,
       scalacheck,
       scalatestplus
-    ),
-    coverageMinimum       := 80,
-    coverageFailOnMinimum := true
+    )
   )
 
-lazy val tests = createModule("tests")
-  .settings(publishArtifact := false)
+lazy val tests = createSubModule("tests")
+  .enablePlugins(NoPublishPlugin)
+  .dependsOn(caffeine, memcached, redis, circe)
+
+lazy val unidoc = createSubModule("unidoc")
+  .enablePlugins(TypelevelUnidocPlugin)
   .dependsOn(caffeine, memcached, redis, circe)
 
-lazy val docs = createModule("docs")
-  .enablePlugins(MicrositesPlugin)
+lazy val docs = createSubModule("docs")
+  .enablePlugins(TypelevelSitePlugin)
   .settings(
-    publishArtifact      := false,
-    micrositeName        := "ScalaCache",
-    micrositeAuthor      := "Chris Birchall",
-    micrositeDescription := "A facade for the most popular cache implementations, with a simple, idiomatic Scala API.",
-    micrositeBaseUrl     := "/scalacache",
-    micrositeDocumentationUrl := "/scalacache/docs",
-    micrositeHomepage         := "https://github.com/cb372/scalacache",
-    micrositeGithubOwner      := "cb372",
-    micrositeGithubRepo       := "scalacache",
-    micrositeGitterChannel    := true,
-    micrositeTwitterCreator   := "@cbirchall",
-    micrositeShareOnSocial    := true,
-    mdocIn                    := (sourceDirectory in Compile).value / "mdoc"
+    tlSiteApiModule := Some((unidoc / projectID).value),
```

Comment:
The `TypelevelUnidocPlugin` sets this one automatically 👍

Reply:
Oh nice, also I had no idea about that `SettingKey` trick that you have used to do it, thanks!